src/wp-includes/formatting.php

```diff
@@ -5390,6 +5390,24 @@ function normalize_whitespace( $str ) {
  * @return string The processed string.
  */
 function wp_strip_all_tags( $string, $remove_breaks = false ) {
+	if ( is_null( $string ) ) {
+		return '';
+	}
```

Comment:
There are cases when `null` has been passed to `wp_strip_all_tags()` (see [this run](https://github.com/WordPress/wordpress-develop/runs/8005398324?check_suite_focus=true#step:17:223)). To maintain BC, this change returns an empty string, without throwing `_doing_it_wrong()`.

https://3v4l.org/BhlOC#v7.4.30 displays the prior behaviour when `null` is passed.

Reply:
Note: This change also protects against the `Passing null to parameter #3 ($subject) of type array|string is deprecated` deprecation in PHP 8.1.

Reply:
If this change is supported, I'll add a test using `null` to protect the behaviour going forward.

Reply:
> However, we must decide whether passing `null` should be allowed for BC, or whether it's `_doing_it_wrong()` and this change just hides the problem.

Based on a super quick review of the plugin repo, it seems [fairly common to pass user input to the function](https://wpdirectory.net/search/01GB92VT6K9Z2C5D7CTSHVBACZ). 

Given the ease with which user submitted data can be turned in to an array by the user and the frequency with which is normalized to a non string by developers `$thing = ! empty( $_POST['t'] ) ? wp_unslash( $_POST['t'] ) : false;`

I'd probably avoid a doing it wrong call because it would fill a site owner's log with a lot of noise they have little control over.

Reply:
Thanks Peter! I've opened a new PR with an alternate approach after your comment and the discussion we had in Slack based on the above comment.

Alternate approach PR: https://github.com/WordPress/wordpress-develop/pull/3133